extensions/azurePublish/src/components/azureProvisionDialog.tsx

```diff
@@ -306,10 +305,13 @@ export const AzureProvisionDialog: React.FC = () => {
     getType,
     getTokenFromCache,
     isGetTokenFromUser,
+    setExtensionState,
+    getExtensionState,
   } = usePublishApi();
   // set type of publish - azurePublish or azureFunctionsPublish
   const publishType = getType();
   const currentConfig = removePlaceholder(publishConfig);
+  const extensionState = getExtensionState();
```

Comment:
you can do:
```const extensionState = getExtensionState() || defaultExtensionState;```
where your default extension state represents all the values you have used down below when the state doesn't exists, then you can avoid all the ```|| something``` expressions down below

Reply:
good idea, Thanks.